src/controllers/TorusController.ts

```diff
@@ -22,6 +22,8 @@ import {
   FEATURES_PROVIDER_CHANGE_WINDOW,
   getPopupFeatures,
   ICommunicationProviderHandlers,
+  IEhandler,
```

Comment:
why you are using IEHandler here not using JRPCRequest type here?

Reply:
I wrapped loginWithPrivateKey provider handler with createGenericJRPCMiddleware.
createGenericJRPCMiddleware can easily plug in new provider handler but have Interface IEhandler which extends JRPCRequest with { windowId?:string , origin?:string }